src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -863,245 +863,262 @@ public SAMReadGroupRecord getReadGroup() {
-     */
-    public void setReadFailsVendorQualityCheckFlag(final boolean flag) {
-        setFlag(flag, SAMFlag.READ_FAILS_VENDOR_QUALITY_CHECK.flag);
-    }
+     * True if the read is marked as failing filters, e.g. QC or Illumina non-PF.
+     * Returns the value of flag bit 0x200.
+     *  */
+    public boolean isFailingFilters() { return getFlag(SAMFlag.READ_FAILS_VENDOR_QUALITY_CHECK); }
 
     /**
-     * the read is either a PCR duplicate or an optical duplicate.
+     * True if the read is NOT marked as failing filters, false otherwise.
+     * Returns the negation of flag bit 0x200.
      */
-    public void setDuplicateReadFlag(final boolean flag) {
-        setFlag(flag, SAMFlag.DUPLICATE_READ.flag);
-    }
+    public boolean isPassingFilters() { return !isFailingFilters(); }
+
+    /** Sets whether the read fails filters. Sets flag bit 0x200. */
+    public void setFailingFilters(final boolean fail) { setFlag(SAMFlag.READ_FAILS_VENDOR_QUALITY_CHECK, fail); }
+
+    /** @deprecated Since May 2017; use {@link #isFailingFilters()} instead. */
+    @Deprecated public boolean getReadFailsVendorQualityCheckFlag() { return isFailingFilters(); }
+    /** @deprecated Since May 2017; use {@link #setFailingFilters(boolean)} instead. */
+    @Deprecated public void setReadFailsVendorQualityCheckFlag(final boolean flag) { setFailingFilters(flag); }
+
 
     /**
-     * Tests if this record is either a secondary and/or supplementary alignment;
-     * equivalent to {@code (getNotPrimaryAlignmentFlag() || getSupplementaryAlignmentFlag())}.
-     */
-    public boolean isSecondaryOrSupplementary() {
-        return getNotPrimaryAlignmentFlag() || getSupplementaryAlignmentFlag();
-    }
+     * True if the read is marked as a duplicate read, false otherwise.
+     * Returns the value of flag bit 0x400. */
+    public boolean isDuplicate() { return getFlag(SAMFlag.DUPLICATE_READ); }
 
-    private void setFlag(final boolean flag, final int bit) {
-        if (flag) {
-            mFlags |= bit;
-        } else {
-            mFlags &= ~bit;
-        }
+    /** Sets whether the read is marked as a duplicate or not. Sets flag bit 0x400. */
+    public void setDuplicate(final boolean duplicate) { setFlag(SAMFlag.DUPLICATE_READ, duplicate); }
+
+    /** @deprecated Since May 2017; use {@link #isDuplicate()} instead. */
+    @Deprecated public boolean getDuplicateReadFlag() { return isDuplicate(); }
+    /** @deprecated Since May 2017; use {@link #setDuplicate(boolean)} instead. */
+    @Deprecated public void setDuplicateReadFlag(final boolean flag) { setDuplicate(flag); }
+
+
+    /** Returns true if the record represents either a secondary or supplementary alignment. */
+    public boolean isSecondaryOrSupplementary() { return isSecondaryAlignment() || isSupplementaryAlignment(); }
+
+    /** Sets a specific flag bit to the provided value. */
+    private void setFlag(final SAMFlag flag, final boolean value) {
```

Comment:
To have more control and do not rely on the new syntax, could this and the getter be public?